operator/pkg/kube/operator/operator.go

```diff
@@ -0,0 +1,240 @@
+		AddFunc: func(obj interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			log.Infof("Processing add: %s", key)
+			if err == nil {
+				queue.Add(key)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			log.Infof("Processing delete: %s", key)
+			if err == nil {
+				queue.Add(key)
+			}
+		},
+	})
+
+	return controller
+}
+
+// Run starts the controller until it receves a message over stopCh.
+func (c *Controller) Run(stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	go c.informer.Run(stopCh)
+
+	// Wait for the caches to be synced before starting workers
+	log.Info("Waiting for informer caches to sync")
+	if !cache.WaitForCacheSync(stopCh, c.informer.HasSynced) {
+		utilruntime.HandleError(fmt.Errorf("timed out waiting for caches to sync"))
+		return
+	}
+	wait.Until(c.runWorker, 5*time.Second, stopCh)
+}
+
+// RunWorker executes the processing loop.
+func (c *Controller) runWorker() {
+	for c.processNextItem() {
+		// continue looping
+	}
+}
+
+// processNextItem retrieves a configmap event from the cache queue.
+func (c *Controller) processNextItem() bool {
+	configMapName, quit := c.queue.Get()
+
+	if quit {
+		return false
+	}
+	defer c.queue.Done(configMapName)
+
+	err := c.processItem(configMapName.(string))
+	if err == nil {
+		// No error, reset the ratelimit counters
+		c.queue.Forget(configMapName)
+	} else if c.queue.NumRequeues(configMapName) < maxRetries {
+		log.Errorf("Error processing %s (will retry): %v", configMapName, err)
+		c.queue.AddRateLimited(configMapName)
+	} else {
+		log.Errorf("Error processing %s (giving up): %v", configMapName, err)
+		c.queue.Forget(configMapName)
+		utilruntime.HandleError(err)
+	}
+
+	return true
+}
+
+// processItem processese a configmap event.
+func (c *Controller) processItem(configMapName string) error {
+	obj, exists, err := c.informer.GetIndexer().GetByKey(configMapName)
+	if err != nil {
+		return fmt.Errorf("error fetching object %s error: %v", configMapName, err)
+	}
+
+	// This looks a little racy...
+	if exists {
+		c.addConfigMap(obj.(*corev1.ConfigMap))
+	} else {
+		c.deleteConfigMap(configMapName)
+	}
+	return nil
+}
+
+// addConfigMap is called in the event that a configmap was added to the system.
+func (c *Controller) addConfigMap(s *corev1.ConfigMap) {
+	log.Info("Converting ConfigMap to CRD object.")
+
+	// yaml.YAMLtoJSON does not appear to understand "---".
+	yamlStrings := strings.Split(s.Data["config"], "---")
+
+	// Build a set of yaml strings and range over them.
+	for _, yamlCRD := range yamlStrings {
+		// Convert the yaml that is split into seprate blobs	
+		jsonBlob, _ := yaml.YAMLToJSON([]byte(yamlCRD))
+
+		// Decode the json into a Kubernetes CRD object
+		crdDeserialized, _, err := deserializer.Decode(jsonBlob, nil, nil)
+		if err != nil {
+			// Coudln't decode the object, no sense creating the CRD
+			fmt.Printf("%v", err)
```

Comment:
Consider using istio logging, to get more info like timestamp etc.

Reply:
agreed.